src/pybind/mgr/progress/module.py

```diff
@@ -409,18 +453,30 @@ def _handle_ls(self):
         else:
             return 0, "", "Nothing in progress"
 
+    def _json(self):
+        return {
+            'events': [ev.to_json() for ev in self._events.values()],
+            'completed': [ev.to_json() for ev in self._completed_events]
+        }
+
     def _handle_clear(self):
         self._events = {}
+        self._completed_events = []
+        self._dirty = True
+        self._save()
+
```

Comment:
might want to clear `self._dirty` here? just to be consistent with `self.serve()` ? or better off, just rename `_save()` to `_maybe_save()`, and do the "check the flag, encode, set_store, clear the flag" dance there?

Reply:
Yep, done.